### DISCUSSION TITLE
Mouse yank has no effect on Windows

### DISCUSSION THREAD

**Discussion by ndtoan96** [helix-editor/helix#8213] [Troubleshooting]
### Summary

https://github.com/helix-editor/helix/assets/33489972/237c3596-faea-440d-8e00-b5c3f885e4bb

### Reproduction Steps

 

I tried this:

1. use the mouse to select some text, see the message selection was yanked to primary clipboard
2. `:primary-clipboard-paste-after`

I expected this to happen: the selected text was pasted after the cursor

Instead, this happened: nothing was pasted

### Helix log

<details><summary>~/.cache/helix/helix.log</summary>

```
please provide a copy of `~/.cache/helix/helix.log` here if possible, you may need to redact some of the lines
```

</details>

### Platform

Windows

### Terminal Emulator

Windows Terminal

### Helix Version

helix 23.05 (7f5940be)

**Comment by the-mikedavis** [comment]
What is the output of `hx --health clipboard`?

**Comment by ndtoan96** [comment]
```
hx --health clipboard
System clipboard provider: clipboard-win
```

I use Windows 11 Home Single Language, version 10.0.22621

**Comment by kirawi** [comment]
I'm pretty sure if a secondary clipboard isn't set then mouse yank is ignored. https://github.com/helix-editor/helix/blob/14401ff75b90cbd98d39eb4b7b5442ad43b49e31/helix-view/src/clipboard.rs#L365

**Reply by ndtoan96** [reply]
Is there any reason for this? If there is then at least we should suppress the yank message too because it causes confusion.
